gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -555,6 +583,7 @@ private void showPolygonInfoDialog(Layer layer) {
                     .getActiveProject()
                     .ifPresentOrElse(
                         project -> {
+                          localValueStore.setPolygonInfoDialogShown(true);
```

Comment:
In Android, in clean architecture in general, Views should never interact directly with data stores. Instead, this should be managed by the ViewModel, which in turn talks to the repository, which talks to the data store. You may need to refactor some of the business logic out of this Fragment into the ViewModel to do this right.

Reply:
updated.